packages/rocknix/sources/scripts/rocknix-lid

```diff
@@ -6,39 +6,136 @@
 
 ### Enable logging
 case $(get_setting system.loglevel) in
-  verbose)
-    DEBUG=true
+verbose)
+  DEBUG=true
   ;;
-  *)
-    DEBUG=false
+*)
+  DEBUG=false
   ;;
 esac
 
 SUSPEND_MODE="$(get_setting system.suspendmode)"
 
 # Only do something when suspend mode is off
 if [ "${SUSPEND_MODE}" != "off" ]; then
-    ${DEBUG} && log $0 "Suspend enabled, exiting"
-    exit 0
+  ${DEBUG} && log $0 "Suspend enabled, exiting"
+  exit 0
 fi
 
+# If a game is running, by default shutdown immediately
+SHUTDOWN_DELAY="$(get_setting system.shutdown_delay)"
+if [ "${SHUTDOWN_DELAY}" = "" ]; then
+    SHUTDOWN_DELAY=0
+fi
+
+# If a game is running, by default delay the shutdown for 5 minutes
+SHUTDOWN_DELAY_RUNNING_GAME="$(get_setting system.shutdown_delay_running_game)"
+if [ "${SHUTDOWN_DELAY_RUNNING_GAME}" = "" ]; then
+    SHUTDOWN_DELAY_RUNNING_GAME=300
+fi
+
+# flag file
+PID=$$
+FLAG_FILE_PATTERN="/var/run/lid-close-shutdown-delay.flag"
+FLAG_FILE="${FLAG_FILE_PATTERN}.${PID}"
+
+# Process to kill
+TO_KILL="rocknix-lid"
+
 # Action = open / close
 ACTION=$1
 
-do_es_shutdown() {
-    # Shutdown only when ES is up and no game is running
-    HTTP_STATUS=$(curl -s -o /dev/null -w "%{http_code}" "http://localhost:1234/runningGame")
-    test $? != 0                 && return 0
-    test "${HTTP_STATUS}" != 201 && return 0 # 201 when not game
+check_es_running_game() {
+  HTTP_STATUS=$(curl -s -o /dev/null -w "%{http_code}" "http://localhost:1234/runningGame")
+  ${DEBUG} && log $0 "ES runningGame HTTP_STATUS - ${HTTP_STATUS}"
+  test $? != 0 && return 1 # call failed, assume no game running
+  test "${HTTP_STATUS}" = 201 && return 1 # 201 when no game running
+  test "${HTTP_STATUS}" = 200 && return 0 # 200 when game running
+}
+
+do_lid_close_actions() {
+  # Turn off display
+  if echo "${UI_SERVICE}" | grep "sway"; then
+    ${DEBUG} && log $0 "Display power off"
+    swaymsg "output * power off"
+  fi
+
+  # Mute audio
+  ${DEBUG} && log $0 "Mute audio"
+  wpctl set-mute @DEFAULT_AUDIO_SINK@ 1
+
+  # CPU core parking
+  ${DEBUG} && log $0 "CPU core parking"
+  for x in /sys/devices/system/cpu/cpu[1-3]*/online; do
```

Comment:
this assume there are only 4 CPUs...
I'd rewrite that to go through all CPUs node, but skip cpu0